src/module.c

```diff
@@ -12210,6 +12210,76 @@ int RM_LoadConfigs(RedisModuleCtx *ctx) {
     return REDISMODULE_OK;
 }
 
+int RM_RdbLoad(const char *filename, int flags) {
+    errno = 0;
+
+    if (flags != 0) {
+        errno = EINVAL;
+        return REDISMODULE_ERR;
+    }
+
+    /* TODO:
+     * - Call fsync for filename
+     * - Stop AOF?
+     * - Kill RDB fork?
+     */
+    emptyData(-1,EMPTYDB_NO_FLAGS,NULL);
+
+    if (server.current_client)
+        protectClient(server.current_client);
+
+    int ret = rdbLoad((char*) filename,NULL,RDBFLAGS_NONE);
+
+    if (server.current_client)
+        unprotectClient(server.current_client);
+
+    /* TODO:
+     * - rename(filename, server.rdb_filename) ?;
+     * - Start AOF?
```

Comment:
yes, we should probably stop and re-start AOF.
regarding the rename, i think that's the module's responsibility to hand over a temp file name and do any rename or even an fsync..
while we're on that subject, maybe we should take in a file descriptor and not a file name?

Reply:
Ok, leaving rename() to the module.